src/components/Spinner/index.tsx

```diff
@@ -3,13 +3,36 @@ import styled from '@emotion/styled'
 import { keyframes } from '@emotion/react'
 import { Spinner as SpinnerIcon } from '../../icons'
 import { color } from '../../theme'
+import { ButtonVariant } from '../Button'
 
 export interface SpinnerProps {
   size?: number
+  variant?: string
 }
 
 const StyledSpinnerIcon = styled(SpinnerIcon)`
-  color: ${color.foreground};
+  color: ${props =>
+    props.variant === ButtonVariant.primary
+      ? color.onAction
+      : props.variant === ButtonVariant.secondary
+      ? color.info
+      : props.variant === ButtonVariant.caution
+      ? color.failure
+      : props.variant === ButtonVariant.success
+      ? color.onSuccess
+      : props.variant === ButtonVariant.warning
+      ? color.onWarning
+      : props.variant === ButtonVariant.danger
+      ? color.onFailure
+      : props.variant === ButtonVariant.facebook
+      ? color.onAction
+      : props.variant === ButtonVariant.inverted
+      ? color.info
+      : props.variant === ButtonVariant.apple
+      ? color.invertedForeground
+      : props.variant === ButtonVariant.google
+      ? color.darkBackground
+      : color.onAction};
```

Comment:
Hmmm, doubt if we should use variants for the spinner. For example, these variants are focused on buttons but we also use a spinner on different pages as a "page loader". What we also can do is keep the `foreground` color by default, and then inside the button component specify a color for each variant. Other than the buttons I can't really name a place where we would use those variants 🤔